.github/workflows/test.yml

```
@@ -0,0 +1,95 @@
+name: Test
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Elixir ${{matrix.elixir}} / OTP ${{matrix.otp}}
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres
+        ports:
+        - 5432:5432
+        env:
+          POSTGRES_DB: paginator_test
+          POSTGRES_PASSWORD: postgres
+
+    strategy:
+      matrix:
+        elixir:
+        - "1.7"
+        - "1.8"
+        - "1.9"
+        - "1.10"
+        - "1.11"
+        otp:
+        - "19.3"
+        - "20.0"
+        - "21.0"
+        - "22.0"
+        - "23.0"
+        exclude:
+        - elixir: 1.8
+          otp: 19.3
+        - elixir: 1.9
+          otp: 19.3
+        - elixir: 1.10
+          otp: 19.3
+        - elixir: 1.10
+          otp: 20.0
+        - elixir: 1.11
+          otp: 19.3
+        - elixir: 1.11
+          otp: 20.0
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set up Elixir
+      uses: actions/setup-elixir@v1
+      with:
+        elixir-version: ${{ matrix.elixir }}
+        otp-version: ${{ matrix.otp }}
+
+    - name: Restore deps cache
+      uses: actions/cache@v2
+      with:
+        path: deps
+        key: ${{ runner.os }}-mix-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-mix-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+          ${{ runner.os }}-mix-${{ matrix.otp }}-${{ matrix.elixir }}
+          ${{ runner.os }}-mix-${{ matrix.otp }}
+          ${{ runner.os }}-mix
+
+    - name: Restore _build cache
+      uses: actions/cache@v2
+      with:
+        path: _build
+        key: ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+          ${{ runner.os }}-build-${{ matrix.otp }}-${{ matrix.elixir }}
+          ${{ runner.os }}-build-${{ matrix.otp }}
+          ${{ runner.os }}-build
+
+    - name: Install Dependencies
+      run: |
+        mix local.hex --force
+        mix local.rebar --force
+        mix deps.get
```

Suggest running these as separate steps. There's little runtime cost to doing so and it makes debugging easier.
```suggestion
    - name: Install hex
      run: mix local.hex --force

    - name: Install rebar
      run: mix local.rebar --force

    - name: Install package dependencies
      run: mix deps.get
```